core/vm/evm.go

```diff
@@ -294,7 +304,11 @@ func (evm *EVM) CallCode(caller ContractRef, addr common.Address, input []byte,
 		// Initialise a new contract and set the code that is to be used by the EVM.
 		// The contract is a scoped environment for this execution context only.
 		contract := NewContract(caller, AccountRef(caller.Address()), value, gas)
-		contract.SetCallCode(&addrCopy, evm.StateDB.GetCodeHash(addrCopy), evm.StateDB.GetCode(addrCopy))
+		// try get from cache
+		code := evm.StateDB.GetCode(addr)
```

Comment:
```suggestion
		code := evm.StateDB.GetCode(addrCopy)
```

Though the effect might be the same, I think it's better to align the previous pattern.